### DISCUSSION TITLE
Implementation of a splash screen (preloader) in the new App Router.

### DISCUSSION THREAD

**Discussion by geraldokereke** [vercel/next.js#50437] [App Router]
A splash screen is a visually appealing user interface element that adds an aesthetic feel to my application. In the previous versions of Next.js, without the app router, I achieved this by utilizing the useEffect hook with a timeout. By employing a conditional rendering approach, I displayed a full-screen loader component initially and then transitioned to the main app component after a few seconds. However, implementing the same approach in the new app router is not straightforward. The app router's Layout.tsx component is treated more like a server component, limiting the usage of hooks. Is there a recommended workaround to achieve a similar outcome in this scenario?

**Comment by Ymirke** [comment]
I think I understand your question. 

If your loading screen is purely cosmetic the way to achieve the same with the app router is this:

**{route}/page.tsx**
```tsx
export default async function ChatPage() {
  return (
    <>
      {'This is a client component'}
      <FullScreenLoader />
      <div>{'.....'}</div>
    </>
  );
}
```

**{route}/FullScreenLoader.tsx**
```tsx
"use client"; // Indicates to next.js that this component is client side.
import React, { useEffect, useState } from "react";

export default function FullScreenLoader() {
  const [show, setShow] = useState(true);
  useEffect(() => {
    setTimeout(() => {
      setShow(false);
    }, 1000)
  }, []);

  if (!show) return null;

  return (
     'Loader...'
  );
}
```

**Comment by mayank1513** [comment]
Add a client component for your splash screen. In my case, I am simply showing a loader for some time - checkout - https://mayank-chaudhari.vercel.app/

```
// splashScreen.tsx
"use client";

export default function SplashScreen() {
  const splashRef = useRef();
  useEffect(()=>{
    setTimeout(()=>{
        splashRef.current.style.display = 'none';
      }, 500);
  }, []);
  return <div ref={splashRef}>...</div>
}
```

And add this component in your layout. Make sure to add proper styles to keep this on top of all your other stuff.

```
.example-class {
  z-index: 1000;
  position: fixed;
  width: 100vw;
  height: 100vh;
}
```
